pubspec.lock

```diff
@@ -1,13 +1,27 @@
 # Generated by pub
 # See https://dart.dev/tools/pub/glossary#lockfile
 packages:
+  archive:
```

Comment:
I'm not sure where all these pubspec changes came from, but I figured they must be necessary if flutter added them.

Reply:
Based on the specifications in `pubspec.yaml`, `pub` finds a set of as-new-as-possible dependencies and stores it in the `lock` file, "locking" the project to those, so that tiny changes causing incompatibilities among libraries won't break our app.

When `pub upgrade` is (indirectly?) run, this `lock` file is regenerated, and if there are new versions of dependencies where we specify in pubspec.yaml to use the latest version (`foo:` instead of `foo: 1.2.3`), they will be reflected in the new `lock` file. So, yep, the `pubspec.lock` file should be checked in so that everyone's project is pinned to the same version in between moments when `pub upgrade` happens.

Come to think of it I don't manually run that command much, so I don't know exactly what causes our lockfile to ever change. I think something like “using a new Flutter version” seems to force `pub` to recompute the dependencies and regenerate the lock file.

See also: the [docs](https://dart.dev/guides/packages#upgrading-a-dependency).